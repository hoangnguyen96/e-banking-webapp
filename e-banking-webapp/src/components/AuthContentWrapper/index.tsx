'use client';

import { ReactNode } from 'react';
import { usePathname } from 'next/navigation';
import Link from 'next/link';
import Image from 'next/image';

// Constants
import { IMAGES, ROUTES } from '@/constants';

// Interfaces
import { TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Components
import { Text } from '../common';

interface AuthContentWrapperProps {
  textFooter?: string;
  textHeading: ReactNode;
  formContent: ReactNode;
}

export const AuthContentWrapper = ({
  textHeading,
  formContent,
  textFooter,
}: AuthContentWrapperProps) => {
  const pathname = usePathname();

  const isOnSignIn = pathname === ROUTES.SIGN_IN;

  const linkDetails = {
    text: isOnSignIn ? 'Register' : 'Sign In',
    href: isOnSignIn ? ROUTES.SIGN_UP : ROUTES.SIGN_IN,
  };

  const classLogo = isOnSignIn ? 'mt-0 md:mt-[108px]' : '';

  return (
    <div className='mx-auto flex w-full max-w-lg flex-1 flex-col'>
      {/* Logo */}
      <Image
        className={`pb-9 ${classLogo}`}
        src={IMAGES.LOGO}
        width={104}
        height={36}
        alt='Logo EBanking'
      />

      <div className='mb-7 max-w-[406px]'>
        <Text as='h1' size={TEXT_SIZE['4XL']} variant={TEXT_VARIANT.DEFAULT}>
          {textHeading}
        </Text>
      </div>

      {/* Form */}
      <div className='mb-[17px] max-w-[325px]'>{formContent}</div>

      {textFooter && (
        <Text
          size={TEXT_SIZE.SM}
          variant={TEXT_VARIANT.DEFAULT}
          className='leading-[20px]'
        >
          {textFooter}&nbsp;
          <Link
            href={linkDetails.href}
            className='leading-[20px] text-primary-200'
          >
            {linkDetails.text}
          </Link>
        </Text>
      )}
    </div>
  );
};
