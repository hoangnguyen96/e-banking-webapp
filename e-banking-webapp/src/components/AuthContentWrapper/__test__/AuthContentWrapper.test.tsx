import { render } from '@testing-library/react';

// Components
import AuthContentWrapper from '..';

describe('AuthContentWrapper component', () => {
  it('Should render snapshot correctly.', () => {
    const props = {
      formContent: <>Form</>,
      textHeading: 'Good To See You Again!',
      textFooter: 'Donâ€™t have an account?',
    };

    expect(render(<AuthContentWrapper {...props} />)).toMatchSnapshot();
  });
});
