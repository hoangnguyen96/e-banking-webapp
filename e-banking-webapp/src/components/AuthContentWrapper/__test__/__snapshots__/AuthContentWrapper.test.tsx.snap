// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContentWrapper component Should render snapshot correctly. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-1 flex-col"
      >
        <img
          alt="Logo EBanking"
          class="pb-9 "
          data-nimg="1"
          decoding="async"
          height="36"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
          style="color: transparent;"
          width="104"
        />
        <div
          class="mb-7"
        >
          <h1
            class="text-foreground-100 text-4xl font-semibold"
          >
            Good To See You Again!
          </h1>
        </div>
        <div
          class="mb-[17px]"
        >
          Form
        </div>
        <p
          class="text-foreground-100 text-sm font-medium"
        >
          Don’t have an account?
           
          <a
            class="text-primary-200"
            href="/sign-in"
          >
            Login
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-1 flex-col"
    >
      <img
        alt="Logo EBanking"
        class="pb-9 "
        data-nimg="1"
        decoding="async"
        height="36"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
        style="color: transparent;"
        width="104"
      />
      <div
        class="mb-7"
      >
        <h1
          class="text-foreground-100 text-4xl font-semibold"
        >
          Good To See You Again!
        </h1>
      </div>
      <div
        class="mb-[17px]"
      >
        Form
      </div>
      <p
        class="text-foreground-100 text-sm font-medium"
      >
        Don’t have an account?
         
        <a
          class="text-primary-200"
          href="/sign-in"
        >
          Login
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
